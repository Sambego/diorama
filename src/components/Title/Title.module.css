.title {
  margin: 0;
  max-width: 80vw;
  font-size: 6vw;
  font-weight: 700;
  text-transform: uppercase;
}
