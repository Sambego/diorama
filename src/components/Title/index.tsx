import React from "react";
import PropTypes from "prop-types";

import styles from "./Title.module.css";

type TitleProps = React.PropsWithChildren<{
	className?: string;
	style?: React.CSSProperties;
}>;

function Title({ children, style, className }: TitleProps) {
	return (
		<h1 style={style} className={`${styles.title} diorama-title ${className}`}>
			{children}
		</h1>
	);
}

Title.propTypes = {
	children: PropTypes.node.isRequired,
	className: PropTypes.string,
	style: PropTypes.shape({}),
};

Title.defaultProps = {
	className: "",
	style: {},
};

export default Title;
