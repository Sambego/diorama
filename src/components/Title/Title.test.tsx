import React from "react";
import renderer from "react-test-renderer";
import Title from "./";

it("renders correctly", () => {
	const tree = renderer.create(<Title>Title content</Title>).toJSON();
	expect(tree).toMatchSnapshot();
});
