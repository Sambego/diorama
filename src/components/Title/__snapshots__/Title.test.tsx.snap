// Vitest Snapshot v1

exports[`renders correctly 1`] = `
<h1
  className="_title_1woga_1 diorama-title "
  style={{}}
>
  Title content
</h1>
`;
